lib/secret-storage/nondumpable-allocator.c

```diff
@@ -94,11 +94,11 @@ static gpointer
 _mmap(gsize len)
 {
   gpointer area = mmap(NULL, len, PROT_READ|PROT_WRITE, MAP_ANONYMOUS|MAP_PRIVATE, -1, 0);
-  if (!area)
+  if (area == (void *)-1)
```

Comment:
man says the return value on error is `MAP_FAILED` from `<sys/mman.h>`.
It is (void *)-1, but this is a implementation detail. Please use `MAP_FAILED` if possible.